packages/gatsby/src/commands/develop-process.ts

```diff
@@ -232,10 +259,48 @@ module.exports = async (program: IProgram): Promise<void> => {
           (_, { data }): DataLayerResult => data
         ),
       },
-    }).withContext({ program, parentSpan: bootstrapSpan, app })
+    }).withContext({ program, parentSpan: bootstrapSpan, app, firstRun: true })
   )
+
+  const isInterpreter = <T>(
+    actor: Actor<T> | Interpreter<T>
+  ): actor is Interpreter<T> => `machine` in actor
+
+  const listeners = new WeakSet()
+  let last: State<IBuildContext, AnyEventObject, any, any>
+
   service.onTransition(state => {
-    reporter.verbose(`transition to ${JSON.stringify(state.value)}`)
+    if (!last) {
```

Comment:
Could you explain what this function does and why we need listeners? This looks bad atm.

Reply:
This is just so we can log the state transitions